docs/meps/mep-0002.md

```diff
@@ -0,0 +1,368 @@
+We have omitted the auto-link syntax from the overview for brevity, they are shown in detail below.
+In all cases, the existing role syntax should continue to work and receive ongoing support from the parser(s).
+
+### Syntax
+
+The parts of the link are `[text](link "title")` with an optional scheme (`[text](scheme:link "title")`).
+The `"title"` is not modified in our proposed syntax. Auto Link syntax, `<scheme:link>`, requires the scheme to be present, we follow the CommonMark [definition of a scheme](https://spec.commonmark.org/0.30/#scheme).
+
+#### `text`
+
+If the `text` is not included, it will be filled in by the default of the target.
+
+- If the reference target is enumerated this will by default be "Sec 2.3"; "Fig. 1" for enumerated figures; "(5)" for equations; "Thm. 2", "Lemma 3", etc. depending on the target type. This can be customized in project configuration or frontmatter, however, the details of that are out of scope for this MEP.
+- If the reference target is not enumerated, the title will be used. This may be the section header text (including syntax/style) or a figure caption.
+- If the node is not enumerated and does not have a title, the reference label will be used.
+
+If the `text` is included it will be used as is with two additional template values (`{number}` and `{name}`)
+
+- Enumeration (`{number}`)
+
+  - Any reference target that is enumerated can reference that number or string with a `{number}`; the `{number}` is only the enumerator and does not include any preceding text.
+    - For example, if referencing `#my-section` with default text of `Section 2.1.2`, you can change the reference to `See 2.1.2` with `[See {number}](#my-section)`.
+  - If a `{number}` is used and the node is not enumerated, the `{number}` will be replaced by "??" and a warning raised.
+
+- Reference text (`{name}`)
+  - Any node can include the text of a reference including any styles (e.g. Sections are the section title; Figures and Tables are the caption).
+  - If `{name}` is used and the node does not have an explicit name or title, the node reference label will be used.
+
+In both cases, the template can be escaped with a preceding backslash, that is `\{number}` or `\{name}`, and the text will not be replaced.
+
+#### `link`
+
+The links are defined by a scheme, which can be standard protocols (`http:`, `mailto:`).
+Here we propose two new schemes, `project` and `path`:
+the `project` scheme allows for cross-referencing pages, sections, equations, figures, or other components of a MyST project;
+the `path` scheme allows for referencing files outside of the project or explicitly downloading the source of a document in the project.
+The schemes are an extensibility point specifically described by [CommonMark](https://spec.commonmark.org/0.30/#example-598)
+and are used to indicate that the link should be resolved by MyST specific logic, and follows standard [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) syntax:
```

Comment:
Just a tiny grammar comment, otherwise the new text seems good.

```suggestion
and are used to indicate that the link should be resolved by MyST specific logic. They follow standard [URI](https://en.wikipedia.org/wiki/Uniform_Resource_Identifier) syntax:
```

Reply:
Thanks. :)